### PR TITLE
🐛Destination-Bigquery: updated check method to handle more possible config issues

### PR DESCRIPTION
**Pull request by etsybaev**
## What
The BigQuery destination doesn't try to do a test write to the schema directly set by the user, but only to tmp one.

## !!! TODO !!! Bump versions and update the changelog once the PR gets approved

Found another weak place in the check method where we check that schema exists but don't try to write anything to it. 

## How
Improved check method to handle more config issues.

## 🚨 User Impact 🚨
No impact is expected for already existing users

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java [hunk 1]
```diff
@@ -88,12 +88,16 @@ public AirbyteConnectionStatus check(final JsonNode config) {
...
 
+      // check has permission for custom dataset creation (may be required for normalization)
       BigQueryUtils.checkHasCreateAndDeleteDatasetRole(bigquery, datasetId, datasetLocation);
 
       final Dataset dataset = BigQueryUtils.getOrCreateDataset(bigquery, datasetId, datasetLocation);
       if (!dataset.getLocation().equals(datasetLocation)) {
         throw new ConfigErrorException("Actual dataset location doesn't match to location from config");
       }
+      // check permission for provided dataset
+      BigQueryUtils.attemptCreateTableAndTestInsert(bigquery, datasetId);
```
Comment (ryankfu) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java:99]: Isn't this method already called in `checkHasCreateAndDeleteDatasetRole`? How come we're re-using this method after it's already called above?

Comment (grishick) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java:99]: +1. `checkHasCreateAndDeleteDatasetRole` calls `attemptCreateTableAndTestInsert`, so I am not sure what adding it here does

**Review by grishick — CHANGES_REQUESTED** [review]
Based on the details that @etsybaev spelled out in the issue https://github.com/airbytehq/airbyte/issues/19998, it does not look like this change is actually addressing the problem. The solution that is needed in addition to https://github.com/airbytehq/airbyte/issues/20561 and https://github.com/airbytehq/airbyte/issues/21030 is to throw config error with a message that explains what the user should do to fix the situation.

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java (hunk 1 above)
Comment (etsybaev) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java:99]: Yup. It's, but the first time we try to create a new table and insert it to a newly created schema meanwhile in this PR I added calls to check it in the selected schema itself

Comment (grishick) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java:99]: I don't think I understand how that is so. This line `BigQueryUtils.attemptCreateTableAndTestInsert(bigquery, datasetId); ` uses the same `datasetId` as this line `BigQueryUtils.checkHasCreateAndDeleteDatasetRole(bigquery, datasetId, datasetLocation);`, so how are they going to be checking different schemas? AFAIK, `config` object that is passed to `check` method contains the default destination configuration and if the user chooses "Mirror source structure" or "Custom format", this datasetId is not going to be the same datasetId as the one where the destination will attempt writing during the sync job. What am I missing here?

Comment (etsybaev) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java:99]: In the already existing verification, we add a suffix 
 final String tmpTestDatasetId = datasetId + CHECK_TEST_DATASET_SUFFIX  + System.currentTimeMillis();

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryUtils.java#L128

So basically in the master branch we currently don't try to write anything directly to the schema set by user in connection config. But we create another schema and do all actions in this tmp schema. 

In this PR I added logic to also create tmp table and write to it directly to the schema set by user. So more cases would be tested now

Comment (grishick) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java:99]: > In this PR I added logic to also create tmp table and write to it directly to the **schema set by user**. So more cases would be tested now.

I don't see how this part is true:  **schema set by user**. What this PR does if I read it correctly is try to create a dataset with a random name and then a test table in that dataset. I don't see where it uses a dataset specified by user in replication configuration. Specifically, if a user choses "Mirror source structure" option, this test will not be trying to create tables in the same namespace where the sync will try to create tables.

**Pull request closed without merging**